x-pack/plugins/data_visualizer/public/application/file_data_visualizer/components/edit_flyout/overrides.js

```diff
@@ -586,6 +637,30 @@ function getColumnNames(columnNames, originalSettings) {
   };
 }
 
+function getGrokFieldNames(grokPattern, originalGrokPattern) {
+  if (originalGrokPattern === undefined) {
+    return {
+      newGrokFieldNames: [],
+      originalGrokFieldNames: [],
+    };
+  }
+
+  if (grokPattern === undefined) {
+    const originalGrokFieldNames = getFieldsFromGrokPattern(originalGrokPattern).map((f) => f.name);
+    return {
+      newGrokFieldNames: originalGrokFieldNames,
+      originalGrokFieldNames,
+    };
+  }
+
+  const newGrokFieldNames = getFieldsFromGrokPattern(grokPattern).map((f) => f.name);
+
+  return {
+    newGrokFieldNames,
+    originalGrokFieldNames: newGrokFieldNames,
```

Comment:
It's not super clear at first glance why we are returning `originalGrokFieldNames` or `newGrokFieldNames` twice here if grokPattern is (or not) undefined. Would be good to add a comment explaining the function.

Reply:
yep it's good to question this. it is badly written. `originalGrokFieldNames` is only used when this function is called on initialisation. So really `originalGrokFieldNames` shouldn't be returned from here at all, and instead it should be created on initialisation.

Reply:
Changed in https://github.com/elastic/kibana/pull/122274/commits/fa65aad8c84df07871451d63819ffa35b2f33a6a